Revert boost/bind.hpp changes in header files and bump to 11.10.1

scpeters opened the pull request:
As noted in https://github.com/osrf/gazebo/pull/3156#issuecomment-1012404375, the changes in #3156 to header files have the potential to break downstream software. This pull request reverts the changes to header files and bumps the patch version to 11.10.1, so we can make a quick release and unbreak downstream software.

cc @traversaro (apologies to @alexdewar )

Update: I started wondering if this would affect gazebo_ros_pkgs, since I see `_1` in [gazebo_ros_api_plugin on `noetic-devel`](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/2.9.2/gazebo_ros/src/gazebo_ros_api_plugin.cpp#L158), but it didn't fail its latest CI job (using Ubuntu 20.04):

* [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ros_gazebo_pkgs-ci-default_noetic-focal-amd64&build=24)](https://build.osrfoundation.org/job/ros_gazebo_pkgs-ci-default_noetic-focal-amd64/24/) https://build.osrfoundation.org/job/ros_gazebo_pkgs-ci-default_noetic-focal-amd64/24/

I believe this is because `boost/thread.hpp` provides an opportunistic include of `boost/bind.hpp` for boost 1.72 and earlier (see https://github.com/boostorg/thread/compare/boost-1.72.0...boost-1.73.0#diff-c94867ed785b09487490187250e85510d4ac8698e7fa22dc56906baeb83423d0L33), so this does not affect Ubuntu 20.04, which uses boost 1.71. That's also why the homebrew CI failed in #3156 but not the Ubuntu CI.

So I still support the changes in this pull request, but I've convinced myself that it is not as dire as it sounded, since it does not affect Ubuntu 20.04.

scpeters committed:
Revert boost/bind.hpp changes in header files

Changing the includes can break downstream software,
so let's revert this for now.

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters committed:
Bump to 11.10.1

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters committed:
Fix Changelog links

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

alexdewar commented:
I might have missed something here... Does this revert fix something?

Before the boost::bind changes there wasn't actually anything *broken* downstream (e.g. in the AUR package I maintain), just a lot of log spam from harmless deprecation warnings, but it seemed like a sensible thing to fix.

alexdewar commented:
If this fix no longer works, would it be possible to merge my (now-closed) PR #3147 instead? That would at least get rid of the log spam (which can make it harder to debug other issues).

scpeters committed:
Define BOOST_BIND_GLOBAL_PLACEHOLDERS in headers

Define this symbol to fix warnings in the specific
header files where we include boost/bind.hpp,
which is more targeted than a -D compiler flag.

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters commented:
> If this fix no longer works, would it be possible to merge my (now-closed) PR #3147 instead? That would at least get rid of the log spam (which can make it harder to debug other issues).

does https://github.com/osrf/gazebo/pull/3160/commits/1718514d0218283528f29c9f84ab44846b27937d work for you @alexdewar ?

scpeters committed:
Merge remote-tracking branch 'origin/gazebo11' into revert_boost_bind_header_changes

scpeters committed:
Add #3161 to changelog

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters committed:
Update date for 11.10.1

Signed-off-by: Steve Peters <scpeters@openrobotics.org>
